catroid/src/main/java/org/catrobat/catroid/content/Sprite.java

```diff
@@ -86,7 +86,7 @@
 	public transient PenConfiguration penConfiguration = new PenConfiguration();
 	private transient boolean convertToSingleSprite = false;
 	private transient boolean convertToGroupItemSprite = false;
-	private transient Multimap<EventId, EventThread> idToEventThreadMap = HashMultimap.create();
+	private transient Multimap<EventId, EventThread> idToEventThreadMap = LinkedHashMultimap.create();
```

Comment:
Please let me know why you switched the `HashMultimap` with a `LinkedHashMultimap` :wink:

Reply:
When an event is received, then all eventThreads are started in the order given by idToEventThreadMap. However the HashMultimap type didn't ensure consistent order.

This is especially problematic for testing behaviour of threads during a single tick.
If one thread should pause all other threads, but sometimes it is executed before & sometimes it is executed after the other threads, it's a pain to test.

In short it makes the execution order of the programs deterministic which is nice :-)

Reply:
Got it! Great explanation! Thanks a lot! :bowing_man: